tracing-subscriber/src/filter/subscriber_filters/mod.rs

```diff
@@ -1006,6 +1022,18 @@ impl FilterState {
         }
     }
 
+    /// Run a second filtering pass, e.g. for Subscribe::event_enabled.
+    fn and(&self, filter: FilterId, f: impl FnOnce() -> bool) -> bool {
+        let map = self.enabled.get();
+        let enabled = if self.enabled.get().is_enabled(filter) {
+            f()
+        } else {
+            false
+        };
```

Comment:
tiny style nit, take it or leave it: doesn't this simplify to
```suggestion
        let map = self.enabled.get();
        let enabled = map.is_enabled(filter) && f();
```
since `f()` won't be evaluated if `&&` short-circuits?

i'm fine with keeping the `if` if we think it's meaningfully more explicit, but it should be equivalent...

Reply:
Writing this long-form comes from copying `did_enable` above. Since there's no extra context to provide here (beyond perhaps making the short-circuit behavior painfully obvious), I think I agree it's better to write this as `&&`... at least if there's no `debug_assertions` bookkeeping we want to do here, or perhaps if we want to skip re-setting `self.enabled`. Using `&&` makes it clear that we're implementing `and` properly :)